core/server/master/src/main/java/alluxio/master/meta/MetaMasterMasterClient.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.AbstractMasterClient;
+import alluxio.Constants;
+import alluxio.master.MasterClientConfig;
+import alluxio.thrift.AlluxioService;
+import alluxio.thrift.GetMasterIdTOptions;
+import alluxio.thrift.MasterHeartbeatTOptions;
+import alluxio.thrift.MetaCommand;
+import alluxio.thrift.MetaMasterMasterService;
+import alluxio.thrift.RegisterMasterTOptions;
+import alluxio.wire.ConfigProperty;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A wrapper for the thrift client to interact with the primary meta master,
+ * used by Alluxio standby masters.
+ * <p/>
+ * Since thrift clients are not thread safe, this class is a wrapper to provide thread safety, and
+ * to provide retries.
+ */
+@ThreadSafe
+public final class MetaMasterMasterClient extends AbstractMasterClient {
+  private MetaMasterMasterService.Client mClient = null;
```

Comment:
nit: member variables are `null` by default, we don't set them to null typically.

Reply:
We have to set it to null to avoid the mvn bugs:

MetaMasterMasterClient.mClient not initialized in constructor and dereferenced in alluxio.master.meta.MetaMasterMasterClient.lambda$getId$0(String) [alluxio.master.meta.MetaMasterMasterClient] At MetaMasterMasterClient.java:[line 78]